artemis-journal/src/main/java/org/jctools/maps/NonBlockingHashSetLong.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICLongNSLong-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTILongS OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jctools.maps;
+
+import java.io.Serializable;
+import java.util.AbstractSet;
+import java.util.Iterator;
+
+/**
+ * A simple wrapper around {@link NonBlockingHashMapLong} making it implement the
+ * {@link java.util.Set} interface.  All operations are Non-Blocking and multi-thread safe.
+ */
+public class NonBlockingHashSetLong extends AbstractSet<Long> implements Serializable {
```

Comment:
@michaelandrepearce would be ok to move this under utils, under an artemis package? just in case.. I think that's more clear, especially with osgi usage.